docs/3.0rc/deploy/infrastructure-concepts/deploy-via-python.mdx

```diff
@@ -0,0 +1,385 @@
+---
+title: Deploy flows with Python
+description: Learn how to use the Python SDK to deploy flows to run in work pools.
+---
+
+Prefect offers a flexible way to deploy flows to dynamic infrastructure using the Python SDK. 
+This approach allows you to target specific work pools and utilize dynamically provisioned infrastructure.
+
+Deploying flows to dynamic infrastructure offers a flexible and programmatic approach to managing your 
+Prefect workflows. While `flow.serve` remains a valid and straightforward method for deploying flows to persistent 
+infrastructure, there are scenarios where you might need to leverage dynamic, scalable resources. In these cases, 
+`flow.deploy` provides a powerful alternative that allows you to target specific work pools and utilize dynamically 
+provisioned infrastructure.
+
+## When to consider `flow.deploy` over `flow.serve`
+
+The `flow.serve` method is an excellent choice for many use cases, especially when you have readily available, persistent infrastructure. However, you might want to consider using `flow.deploy` in the following situations:
+
+1. Cost optimization: Dynamic infrastructure can help reduce costs by scaling resources up or down based on demand.
+2. Resource scarcity: If you have limited persistent infrastructure, dynamic provisioning can help manage resource allocation more efficiently.
+3. Varying workloads: For workflows with inconsistent resource needs, dynamic infrastructure can adapt to changing requirements.
+4. Cloud-native deployments: When working with cloud providers that offer serverless or auto-scaling options.
+
+By using `flow.deploy`, you can take advantage of Prefect's work pool system, which provides a layer of abstraction between your flows and the underlying infrastructure. This approach offers greater flexibility and efficiency in running your workflows when dynamic infrastructure is needed.
+
+Let's explore how to create a deployment using the Python SDK and leverage dynamic infrastructure through work pools.
```

Comment:
Maybe note that you're going to use Docker below (maybe because most infrastructure uses it - it's representative) and that there are other options for users depending upon their needs and link to those.

Reply:
Added in https://github.com/PrefectHQ/prefect/pull/15140/commits/dcea3e76240519e04f85572bf2f5df17035ad627!